### PR TITLE
Adding govet

### PR DESCRIPTION
**Pull request by soltysh**
@bparees this is post #276 I'm adding govet, though it'll not fail yet with errors, similarly to what we do with golint. I've also cleaned the `Makefile` a bit to follow origin patterns.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,26 @@ all build:
...
 
+
+# Verify code is properly organized.
+#
+# Example:
+#   make verify
+verify: build
+	hack/verify-gofmt.sh
+	hack/verify-golint.sh || true
+	hack/verify-govet.sh || true
```
Comment (bparees) [inline Makefile]: why || true?  what's the point of the verify target if it doesn't fail when things don't verify?

File: Makefile [hunk 2]
```diff
@@ -36,7 +56,9 @@ all build:
 #   make check
 #   make test
 #   make check WHAT=pkg/build GOFLAGS=-v
-check test:
+check-test: export STI_COVER= -cover -covermode=atomic
+check-test: export STI_RACE=  -race
+check-test: verify
```
Comment (bparees) [inline Makefile]: why'd you change the name of the target?

Comment (bparees) [inline Makefile]: why not just put STI_RACE and STI_COVER inline w/ the hack/test-go invocation?  why 3 targets?

File: Makefile (hunk 1 above)
Comment (soltysh) [inline Makefile]: Once I clean up the errors, I'll remove that.

Comment (soltysh) [inline Makefile]: But that will be a separate PR.

File: Makefile (hunk 2 above)
Comment (soltysh) [inline Makefile]: I've copied that solution from origin/k8s, I'm not a makefile expert but a bit of googling gave me [Target-specific Variable Values](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html), that's why we have separate for each value. But you're right the target name shouldn't be changed.

File: Makefile [hunk 3]
```diff
@@ -36,7 +56,9 @@ all build:
 #   make check
 #   make test
 #   make check WHAT=pkg/build GOFLAGS=-v
-check test:
+check test: export STI_COVER= -cover -covermode=atomic
+check test: export STI_RACE=  -race
+check test: verify
```
Comment (bparees) [inline Makefile]: I still dont' see why you don't just write:  
check test:
  STI_COVER=" - cover -covermode=atomic" STI_RACE=" -race" hack/test-go.sh $(WHAT) $(TESTS)

the example you linked to doesn't "export" the env variable and it's not clear you need to?

Comment (bparees) [inline Makefile]: also your example is for Makefile variables, not shell variables... but you want STI_COVER to be a shell variable...right?

Comment (soltysh) [inline Makefile]: I guess you're right. Fixed.

Comment (soltysh) [inline Makefile]: At the same time, it also means we should fix that in origin, then.

**Comment by bparees** [comment]
lgtm.. merge?

**Comment by soltysh** [comment]
[merge]
